opam-ci-check/lib/lint.ml

```diff
@@ -172,8 +171,20 @@ module Checks = struct
     | None | Some [] -> []
     | Some _ -> [ (pkg, ExtraFiles) ]
 
-  let get_dune_project_version ~pkg url =
-    D.with_temp_dir "lint-dune-project-version-" @@ fun dir ->
+  let parse_dune_project_version dune_project =
+    match
+      In_channel.input_all
+      |> In_channel.with_open_text dune_project
+      |> Sexplib.Sexp.parse
+    with
+    | exception Sys_error _ -> Ok None
+    | Sexplib.Sexp.Done (List [ Atom "lang"; Atom "dune"; Atom version ], _) ->
+        Ok (Some version)
+    | Done _ -> Error "(lang dune ...) is not the first construct"
+    | Cont _ -> Error "Failed to parse the dune-project file"
+
+  let pull_pkg_get_version dir (pkg, url) =
+    let dir = Fpath.to_string dir in
```

Comment:
Is this change related to the addition of the build-test functionality?

Reply:
It's not. I was trying to remove some unnecessary helpers since we've already started using `Bos` in `opam-ci-check`. But, I'll send it as a different PR.